test/framework/src/main/java/org/elasticsearch/cluster/version/CompatibilityVersionsUtils.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.version;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.test.TransportVersionUtils;
+
+public class CompatibilityVersionsUtils {
+
+    /**
+     * Current compatibility versions known at compile time
+     *
+     * <p>Some of our compatibility versions may be constructed at runtime, but in
+     * many tests those will not be needed. This utility method returns only the current values for
+     * compatibility versions defined at runtime, which is generally just those defined in the server
+     * module.
+     * @return Compatibility versions known at compile time.
+     */
+    public static CompatibilityVersions compileTimeCurrent() {
```

Comment:
The system index mappings versions are only determined at runtime, because they depend on which plugins are present. I'm not sure how I could return sensible testing defaults for those unless I hardcode some system index names. (I could reference the tasks and synonyms indices here, for example, but not the security index.)

Reply:
Sounds good!